launcher/firstLaunch/firstlaunch_moc.cpp

```diff
@@ -279,7 +293,69 @@ QString FirstLaunchView::getHeroesInstallDir()
 	return QString{};
 }
 
-void FirstLaunchView::copyHeroesData(const QString & path)
+void FirstLaunchView::extractGogData()
+{
+#ifndef VCMI_NO_INNOEXTRACT
+	QString filterExe = tr("GOG executable") + " (*.exe)";
+	QString fileExe = QFileDialog::getOpenFileName(this, tr("Select a GOG installer (exe) file..."), QDir::homePath(), filterExe);
+	if(fileExe.isEmpty())
+		return;
+
+	QString filterBin = tr("GOG bin file") + " (*.bin)";
+	QString fileBin = QFileDialog::getOpenFileName(this, tr("Select a GOG data (bin) file..."), QFileInfo(fileExe).absolutePath(), filterBin);
+	if(fileBin.isEmpty())
+		return;
+
+	if(QFileInfo(fileExe).completeBaseName() + QString("-1") != QFileInfo(fileBin).completeBaseName())
+	{
+		QMessageBox::critical(this, tr("File name error!"), tr("The filename of exe and bin has to be compatible!"), QMessageBox::Ok, QMessageBox::Ok);
+		return;
+	}
+
+	QTimer::singleShot(100, this, [this, fileExe, fileBin](){ // background to make sure FileDialog is closed...
+		QTemporaryDir dir;
+		if(dir.isValid() && QFile::exists(fileExe) && QFile::exists(fileBin)) {
+			ui->pushButtonGogInstall->setText(tr("Installing... Please wait!"));
```

Comment:
Perhaps also block button while process is ongoing? Or even block whole UI if Qt allows that.

Reply:
Yes the UI is already completely blocked. Not the nicest way - async would be great (as the process needs some seconds). But I think okay for now.